### PR TITLE
fix: use copy builtin correctly for security groups

### LINKED ISSUE #1151: AdditionalSecurityGroups are not applied
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

Have AWSMachine with additionalSecurityGroups specified. Were not applied.

**What did you expect to happen:**

Security groups applied.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

/assign
/lifecycle active

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by rudoi**
**What this PR does / why we need it**:

Turns out, the destination in the copy builtin is first!

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1151

### CODE REVIEW

File: pkg/cloud/services/ec2/instances.go [hunk 1]
```diff
@@ -622,6 +622,8 @@ func (s *Service) attachSecurityGroupsToNetworkInterface(groups []string, interf
 		return nil
 	}
 
+	s.scope.V(3).Info("updating security groups", "groups", totalGroups)
```
Comment (vincepri) [inline pkg/cloud/services/ec2/instances.go]:
```suggestion
	s.scope.V(3).Info("Updating security groups", "groups", totalGroups)
```

Comment (ncdc) [inline pkg/cloud/services/ec2/instances.go]: Is V3 appropriate or would it be worthwhile to log this all the time?

Comment (rudoi) [reply pkg/cloud/services/ec2/instances.go]: there are 2 other relevant logging messages that wrap this, but not with this much detail. i'm open to either.

Comment (ncdc) [reply pkg/cloud/services/ec2/instances.go]: I'd probably err on the side of always logging, and do a follow-up to increase the V level if we find it too verbose when it's always on

Comment (rudoi) [reply pkg/cloud/services/ec2/instances.go]: i think always would be better

**Comment by detiber** [comment]
Need to make sure we include the permissions updates in the release notes

**Comment by ncdc** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1146 is listed as ✨ but we should change it to ⚠️, right?

**Comment by ncdc** [comment]
LGTM. Can you squash this to 2 commits - 1 for the permissions update, 1 for the copy + logging fix?

**Pull request merged**